src/robotFactory.js

```diff
@@ -1,10 +1,88 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: 0,
+      y: 0,
+    };
+  }
 
-class DeliveryDrone {}
+    for (const coord in coords) {
+      if (coords[coord] === undefined) {
+        result[coord] = 0;
+      }
+
+      result[coord] = coords[coord];
+    }
+
+    return result;
+  }
+
+  getInfo() {
+    return (`Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`);
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(...args) {
+    super(...args);
+    
+    if (this.coords.z === undefined) {
+      this.coords.z = 0;
+    }
```

Comment:
do the same for x and y in BaseRobot

```suggestion
    this.coords.z = coords.z || 0;
```